src/js/app.js

```diff
@@ -0,0 +1,196 @@
+      return input;
+    };
+  })
+  .directive('fileUpload', ['alerts', (alerts) => {
+    /*
+     * template to be rendered
+     */
+    const template = `<input type="file" id="file" name="files" multiple>
+                      <label for="file">
+                      <i class="fa fa-upload" aria-hidden="true"></i> 
+                      Upload A file <span>{{ uploadedFilesCount }}
+                      </span></label>`;
+
+
+    /*
+     * link funtion to be called
+     */
+    function link(scope, elem) {
+      elem.on('change', (evt) => {
+        const fileList = evt.target.files;
+        angular.forEach(fileList, (file) => {
+          if (!scope.isValidFile(file.name)) {
+            scope.$apply(function () {
+              alerts("please all file must be a valid json",
+                "danger", true, 5000);
+            });
+            return;
+          }
+          const reader = new FileReader();
+          //event fired when reader.readAsTex is called
+          reader.onload = function (event) {
+            try {
+              const result = JSON.parse(event.target.result);
+              scope.$apply(() => {
+                scope.uploadedFiles[file.name] = result;
+              });
+            } catch (e) {
+              scope.$apply(() => {
+                alerts("invalid json. pls refer to index guide ",
+                  "danger", true, 5000);
+              });
+              return;
+            }
+          };
+          reader.readAsText(file);
+        });
+      });
+    }
+    return {
+      template: template,
+      link: link
+    };
+  }])
+  .controller('mainController', ['$scope', 'alerts', ($scope, alerts) => {
+    /**
+     *  {Object} instanciates the invertedIndex class
+     */
+    const indexer = new InvertedIndex();
+
+    /**
+     *  {Object}  uploaded files
+     */
+    $scope.uploadedFiles = {};
+
+    /** 
+     *  {Object} indexed value to display
+     */
+    $scope.index = [];
+
+    /** 
+     *  {Boolean} shwows indexed table
+     */
+    $scope.showTable = false;
+
+    /** 
+     *  {Array} Holds number of data in a file
+     */
+    $scope.fileCount = [];
+
+    /** 
+     *  {Array} Holds all indexed files
+     */
+    $scope.allIndexedFiles = [];
+
+
+    /** set the value to be selected
+     * @param {String} dom element to get
+     * @return {String} selected value
+     */
+    $scope.selected = (id) => {
+      return document.getElementById(id || 'indexed-files').value;
+    };
+
+
+    /**
+     * @param  {String} fileName - file name
+     * @param  {Object} data  - data to transform
+     * @param  {count} count - count of object in the file
+     */
+    $scope.transformData = (fileName, data, count) => {
+      return {
+        fileName: fileName,
+        data: data,
+        count: count
+      };
+    };
+
+    /**
+     * creates an inverted Index 
+     */
+    $scope.createIndex = () => {
+      const selected = $scope.selected('uploaded-files');
```

Comment:
`const` is for values variables that won't be reassigned throughout your codebase. So in situations where you are going to variable values changing then you should use `let` instead